lib/ASRouterTriggerListeners.jsm

```diff
@@ -24,11 +24,28 @@ this.ASRouterTriggerListeners = new Map([
     _triggerHandler: null,
     _hosts: null,
 
+    /**
+     * Wait for browser startup to finish to avoid accessing uninitialized
+     * properties
+     */
+    async _checkStartupFinished(win) {
+      if (!win.gBrowserInit.delayedStartupFinished) {
+        await new Promise(resolve => {
+          let delayedStartupObserver = () => {
+            Services.obs.removeObserver(delayedStartupObserver, "browser-delayed-startup-finished");
+            resolve();
+          };
+
+          Services.obs.addObserver(delayedStartupObserver, "browser-delayed-startup-finished");
+        });
+      }
+    },
+
     /*
      * If the listener is already initialised, `init` will replace the trigger
      * handler and add any new hosts to `this._hosts`.
      */
-    init(triggerHandler, hosts = []) {
+    async init(triggerHandler, hosts = []) {
       if (!this._initialized) {
```

Comment:
Do we need to worry about init being called multiple times? Potentially multiple concurrent `init`s are waiting for startup to finish because `_initialized` is set at the end.

I suppose ASRouter's `await ASRouterTriggerListeners.get(trigger.id).init(…)` should guard this even if the trigger id repeats as it'll wait.